vllm/v1/core/sched/scheduler.py

```diff
@@ -425,6 +425,18 @@ def schedule(self) -> SchedulerOutput:
             grammar_bitmask=grammar_bitmask,
         )
 
+        # Advance the number of computed tokens for the request AFTER
+        # the request is scheduled.
+        # 1. The scheduler_output of the current step has to include the
+        #    original number of scheduled tokens to determine input IDs.
+        # 2. Advance the number of computed tokens here allowing us to
+        #    schedule the prefill request again immediately in the next
+        #    scheduling step.
+        # 3. If some tokens (e.g. spec tokens) are rejected later, the number of
+        #    computed tokens will be adjusted in update_from_output.
+        for req_id, num_scheduled_token in num_scheduled_tokens.items():
+            self.requests[req_id].num_computed_tokens += num_scheduled_token
```

Comment:
Just wondering: What if we give this updated value to the model runner instead of the pre-updated `num_computed_tokens`? I'm wondering because I'd like to minimize the discrepancy between the scheduler states and the model runner states.

Reply:
That's another option, but I'm not sure if this will need to change lots of places in the model runner. Basically all places use `num_computed_tokens` in the model runner now have to be changed to `num_computed_tokens - num_scheduled_tokens`. I can take a look to see how many places we need to change exactly if you prefer tho

Reply:
I looked around the model runner (mainly at persistent batch and prepare input) and here are some potential changes:
```
num_new_tokens = (num_computed_tokens -
    len(req_data.new_token_ids) -
    req_state.num_tokens)
```
```
positions_np = self.positions_np[:total_num_scheduled_tokens]
np.add(self.input_batch.num_computed_tokens_cpu[req_indices],
    arange,
    out=positions_np)
```

```
common_prefix_len = min(
    common_prefix_len,
    self.input_batch.num_computed_tokens_cpu[:num_reqs].min())
```
And some other small places.

Do you think we should just change all such places? Or we should introduce another attribute (e.g. `prev_num_computed_tokens`) in the persistent batch?

Reply:
Got it. Let's explore that option in a future PR. I'm good with keep it for now.